lib/cli.js

```diff
@@ -157,17 +157,36 @@ function buildWatcherOptions(options) {
   };
 }
 
-function guardOutputDir(outputDir, removeExisting) {
+function guardOutputDir(outputDir, overwrite) {
   if (fs.existsSync(outputDir)) {
-    if (removeExisting) {
-      rimraf.sync(outputDir);
+    if (overwrite) {
+      if (!isParentDirectory(outputDir)) {
+        console.error(
+          'option --overwrite can not be used if outputPath is a parent directory: ' + outputDir
+        );
+        process.exit(1);
+      }
       return;
     }
     console.error(
       outputDir +
         '/ already exists; we cannot build into an existing directory, ' +
-        'pass --overwrite to auto-delete the output directory'
+        'pass --overwrite to overwrite the output directory'
     );
     process.exit(1);
   }
 }
+
+function isParentDirectory(outputPath) {
```

Comment:
maybe you can use `findup-sync` to detect project root by locating `package.json` ?

Reply:
will check